### ISSUE TITLE
azurerm_backup_policy_vm does not validate that policy_type equals V2 when frequency is set to Hourly

### ISSUE THREAD

**Issue by BrokeAssChewbacca** [hashicorp/terraform-provider-azurerm#17817]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.6

### AzureRM Provider Version

3.16.0

### Affected Resource(s)/Data Source(s)

azurerm_backup_policy_vm

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "3.16.0"
    }
  }
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "rg" {
  name     = "rg"
  location = "eastus"
}

resource "azurerm_recovery_services_vault" "rsv" {
  name                = "rsv"
  location            = "eastus"
  resource_group_name = azurerm_resource_group.rg.name
  sku                 = "Standard"
}

resource "azurerm_backup_policy_vm" "hourly" {
  name                = "hourly"
  resource_group_name = azurerm_resource_group.rg.name
  recovery_vault_name = azurerm_recovery_services_vault.rsv.name
  policy_type         = "V1"

  backup {
    frequency     = "Hourly"
    time          = "21:00"
    hour_interval = 4
    hour_duration = 4
  }

  retention_daily {
    count = 10
  }
}
```

### Debug Output/Panic Output

```shell
azurerm_backup_policy_vm.hourly: Still creating... [7m20s elapsed]
azurerm_backup_policy_vm.hourly: Still creating... [7m30s elapsed]
╷
│ Error: creating/updating Azure Backup Protection Policy "hourly" (Resource Group "rg"): backup.ProtectionPoliciesClient#CreateOrUpdate: Failure responding to request: StatusCode=500 -- Original Error: autorest/azure: Service returned an error. Status=500 Code="CloudInternalError" Message="Microsoft Azure Backup encountered an internal error. Wait for a few minutes and then try the operation again. If the issue persists, please contact Microsoft support."
│ 
│   with azurerm_backup_policy_vm.hourly,
│   on main.tf line 26, in resource "azurerm_backup_policy_vm" "hourly":
│   26: resource "azurerm_backup_policy_vm" "hourly" {
```

### Expected Behaviour

When running a `terraform plan` or `terraform apply` with a backup frequency of `Hourly`, if `policy_type` is not set to `V2` it should error and notify you that `V2` is required when backup frequency is set to `Hourly`.

### Actual Behaviour

`terraform plan` reports that the configuration is valid and presents you with the resources it will create.

`terraform apply` reports that the configuration is valid and if accepted attempts to create the backup_policy_vm. After 5-8 minutes, an error is returned from Azure stating that "Microsoft Azure Backup encountered an internal error".

### Steps to Reproduce

`terraform init`
`terraform plan` or `terraform apply`

### Important Factoids

N/A

**Comment by jiaweitao001** [comment]
Hi @BrokeAssChewbacca , thanks for creating this issue. I've reproduced this and I'll make contact to service team to fix it.

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #17880. As such, I am going to mark this issue as closed.
